functions.R

```diff
@@ -881,3 +891,150 @@ plotter <- function(df, variable = NULL, show_labels = TRUE){
   }
 }
 
+# taking factors with one level in osduhs survey data and imputing 'NO' onto it
+relevel_factor_one_lfs <- function(dat_var) {
+  dat_var <- as.character(dat_var)
+  dat_var[is.na(dat_var)] <- 'NO'
+  return(dat_var)
+}
+
+# restructure data types
+restructure_data_types <- function(temp) {
+  for(i in 1:ncol(temp)) {
+    if(grepl('factor', class(temp[, i]))) {
+      temp[, i] <- as.character(temp[, i])
+    } else {
+      temp[, i] <- as.numeric(temp[, i])
+    }
+  }
+  return(temp)
+}
+
+clean_lfs <- function(temp_clean) {
+  
+  # characters back to factors, and everything else numeric
+  temp_clean <- restructure_data_types(temp_clean)
+  
+  # remove extra columns or columns with too many NAs
+  temp_clean$current_student_status_and_type_of_school.1 <- NULL
+  
+  # if the level of a factor is 1, then that factor only has "yes" coded and should replace NA with "NO"
+  temp_clean$job_seeker_checked_wemployers_directly <- relevel_factor_one_lfs(temp_clean$job_seeker_checked_wemployers_directly)
+  temp_clean$job_seeker_checked_wemployment_agency <- relevel_factor_one_lfs(temp_clean$job_seeker_checked_wemployment_agency)
+  temp_clean$jobseeker_contacted_relatives <- relevel_factor_one_lfs(temp_clean$jobseeker_contacted_relatives)
+  temp_clean$jobseeker_looked_at_ads <- relevel_factor_one_lfs(temp_clean$jobseeker_looked_at_ads)
+  temp_clean$jobseeker_placed_or_answered_ads <- relevel_factor_one_lfs(temp_clean$jobseeker_placed_or_answered_ads)
+  temp_clean$jobseeker_other_methods <- relevel_factor_one_lfs(temp_clean$jobseeker_other_methods)
+  
+  # filter to get ontario (should have done this earlier, but missed it)
+  temp_clean <- temp_clean %>% filter(province == 'Ontario')
+  
+  # recode variables that have sloppy coding - We checked and NAs literally mean the question was not appicable to that person. 
+  # Real NAs - actual missing values - were deleted prior to them giving us this data (from carlton university). This is documented in 
+  # data documentation under the Labour Force Survey.
```

Comment:
The below are examples of where you could use `dplyr::mutate`. It won't make your code any "better" in this case, just shorter/more readable... (since you won't have to type `temp_clean$` so much)